src/lib/hooks/swap/useWrapCallback.tsx

```diff
@@ -0,0 +1,137 @@
+import { ContractTransaction } from '@ethersproject/contracts'
+import { Trans } from '@lingui/macro'
+import { useWETHContract } from 'hooks/useContract'
+import { useAtomValue } from 'jotai/utils'
+import { Field, swapAtom } from 'lib/state/swap'
+import tryParseCurrencyAmount from 'lib/utils/tryParseCurrencyAmount'
+import { useCallback, useMemo, useState } from 'react'
+
+import { WRAPPED_NATIVE_CURRENCY } from '../../../constants/tokens'
+import useActiveWeb3React from '../useActiveWeb3React'
+import { useCurrencyBalances } from '../useCurrencyBalance'
+import useNativeCurrency from '../useNativeCurrency'
+
+export enum WrapInputError {
+  NO_ERROR, // must be equal to 0 so all other errors are truthy
```

Comment:
you should explicitly set this to `= 0` if this is the case.

Reply:
copied from app. I really dislike this usage of the 0 index as a falsy value :|

Reply:
you should still explicitly set this to 0, as the comment indicates that you are relying on that value.